air/src/options.rs

```diff
@@ -108,14 +121,47 @@ impl ProofOptions {
     /// - `grinding_factor` is greater than 32.
     /// - `fri_folding_factor` is not 2, 4, 8, or 16.
     /// - `fri_remainder_max_degree` is greater than 255 or is not a power of two minus 1.
-    #[rustfmt::skip]
     pub const fn new(
         num_queries: usize,
         blowup_factor: usize,
         grinding_factor: u32,
         field_extension: FieldExtension,
         fri_folding_factor: usize,
         fri_remainder_max_degree: usize,
+    ) -> ProofOptions {
+        Self::with_num_partitions(
+            num_queries,
+            blowup_factor,
+            grinding_factor,
+            field_extension,
+            fri_folding_factor,
+            fri_remainder_max_degree,
+            1,
+            1,
+        )
+    }
+
+    /// Returns a new instance of [ProofOptions] struct constructed from the specified parameters.
+    ///
+    /// # Panics
+    /// Panics if:
+    /// - `num_queries` is zero or greater than 255.
+    /// - `blowup_factor` is smaller than 2, greater than 128, or is not a power of two.
+    /// - `grinding_factor` is greater than 32.
+    /// - `fri_folding_factor` is not 2, 4, 8, or 16.
+    /// - `fri_remainder_max_degree` is greater than 255 or is not a power of two minus 1.
+    /// - `num_partitions` is zero or greater than 16.
+    #[rustfmt::skip]
+    #[allow(clippy::too_many_arguments)]
+    pub const fn with_num_partitions(
+        num_queries: usize,
+        blowup_factor: usize,
+        grinding_factor: u32,
+        field_extension: FieldExtension,
+        fri_folding_factor: usize,
+        fri_remainder_max_degree: usize,
+        num_partitions: usize,
+        min_partition_size: usize,
     ) -> ProofOptions {
```

Comment:
I would change it to a "builder" pattern constructor. That is, it takes `Self` as a parameter and modifies it. The signature would look something like this:
```rust
pub const fn with_partitions(
    mut self,
    num_partitions: usize,
    min_partition_size: usize
) -> Self {
    ...
}
```
Then, to create options with partitions, users would be able to do something like this:
```rust
let options = ProofOptions::new(...).with_partitions(...);
```

Reply:
Updated to your proposal